server/src/main/java/org/elasticsearch/index/codec/PerFieldFormatSupplier.java

```diff
@@ -139,20 +143,27 @@ public DocValuesFormat getDocValuesFormatForField(String field) {
         if (useTSDBDocValuesFormat(field)) {
             return tsdbDocValuesFormat;
         }
+        if (useLogsDocValuesFormat(field)) {
+            return logsDocValuesFormat;
+        }
         return docValuesFormat;
     }
 
     boolean useTSDBDocValuesFormat(final String field) {
-        if (excludeFields(field)) {
+        if (excludeFields(field) || excludeMapperTypes(field)) {
             return false;
         }
+        return mapperService != null
+            && mapperService.getIndexSettings().useTimeSeriesDocValuesFormat()
+            && mapperService.getIndexSettings().isES87TSDBCodecEnabled();
+    }
 
-        if (excludeMapperTypes(field)) {
+    boolean useLogsDocValuesFormat(final String field) {
+        if (excludeFields(field) || excludeMapperTypes(field)) {
             return false;
         }
-
         return mapperService != null
-            && mapperService.getIndexSettings().useTimeSeriesDocValuesFormat()
+            && mapperService.getIndexSettings().useLogsDocValuesFormat()
             && mapperService.getIndexSettings().isES87TSDBCodecEnabled();
```

Comment:
Instead of this change, let do:

```
 public DocValuesFormat getDocValuesFormatForField(String field) {
        if (useTSDBDocValuesFormat(field)) {
            if (useLargerBlockSize() {
                return tsdbDocValuesFormat;
            }
        }
        return docValuesFormat;
  }

  boolean useLargerBlockSize() {
    // Check index setting
  }
```

I think this is simpler?